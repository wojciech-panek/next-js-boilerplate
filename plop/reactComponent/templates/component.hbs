import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

