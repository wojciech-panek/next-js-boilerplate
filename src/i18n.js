const addLocaleData = require('react-intl').addLocaleData;
const enLocaleData = require('react-intl/locale-data/en');
const plLocaleData = require('react-intl/locale-data/pl');

const enTranslationMessages = require('./translations/en.json');
const plTranslationMessages = require('./translations/pl.json');

const LOCALES = {
  ENGLISH: 'en',
  POLISH: 'pl',
};

const DEFAULT_LOCALE = LOCALES.ENGLISH;

const appLocales = [
  LOCALES.ENGLISH,
  LOCALES.POLISH,
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages = locale !== DEFAULT_LOCALE
    ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages)
    : {};
  return Object.keys(messages).reduce((formattedMessages, key) => {
    const formattedMessage = !messages[key] && locale !== DEFAULT_LOCALE
      ? defaultFormattedMessages[key]
      : messages[key];
    return Object.assign(formattedMessages, { [key]: formattedMessage });
  }, {});
};

const translationMessages = {
  [LOCALES.ENGLISH]: formatTranslationMessages(LOCALES.ENGLISH, enTranslationMessages),
  [LOCALES.POLISH]: formatTranslationMessages(LOCALES.POLISH, plTranslationMessages),
};

if (typeof window !== 'undefined') {
  addLocaleData(enLocaleData);
  addLocaleData(plLocaleData);

  if (!window.Intl) {
    (new Promise((resolve) => {
      resolve(require('intl'));
    }))
      .then(() => Promise.all([
        require('intl/locale-data/jsonp/en.js'),
        require('intl/locale-data/jsonp/pl.js'),
      ]))
      .catch((err) => {
        throw err;
      });
  }
}

module.exports = {
  LOCALES,
  DEFAULT_LOCALE,
  appLocales,
  translationMessages,
};
