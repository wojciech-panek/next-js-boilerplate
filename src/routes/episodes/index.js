export { default as Episodes } from './episodes.container';
