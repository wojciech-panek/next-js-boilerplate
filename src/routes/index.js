const index = module.exports = require('next-routes')();
const i18n = require('../i18n');

const appLocalesWithoutDefault = i18n.appLocales.filter((locale) => locale !== i18n.DEFAULT_LOCALE);
let langParam = '';
if (appLocalesWithoutDefault.length) {
  langParam += `/:lang(${appLocalesWithoutDefault.join('|')})?`;
}

index
  .add('home', `${langParam}`)
  .add('seriesList', `${langParam}/series`)
  .add('series', `${langParam}/series`);
