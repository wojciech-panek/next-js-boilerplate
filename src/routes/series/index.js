export { default as Series } from './series.container';
