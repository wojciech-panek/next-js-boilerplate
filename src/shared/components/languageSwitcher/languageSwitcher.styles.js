import styled from 'styled-components';


export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 20px 0;
`;

export const Select = styled.select`

`;
