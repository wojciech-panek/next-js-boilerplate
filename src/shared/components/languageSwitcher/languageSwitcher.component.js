import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Select, Container } from './languageSwitcher.styles';
import { appLocales, DEFAULT_LOCALE } from '../../../i18n';
import { Router } from '../../../routes';


export class LanguageSwitcher extends PureComponent {
  static propTypes = {
    language: PropTypes.string.isRequired,
    router: PropTypes.object.isRequired,
  };

  handleChange = (e) => {
    const currentLanguageCode = this.props.router.query.lang ? `/${this.props.router.query.lang}` : '';
    const targetLanguageCode = e.target.value === DEFAULT_LOCALE ? '' : `/${e.target.value}`;
    let targetUrl = this.props.router.asPath.replace(currentLanguageCode, targetLanguageCode).replace(/\/$/, '');

    if (!targetUrl.length) {
      targetUrl = '/';
    }

    Router.pushRoute(targetUrl);
  };

  render() {
    return (
      <Container>
        <Select value={this.props.language} onChange={this.handleChange}>
          {appLocales.map((locale, index) => (
            <option key={index} value={locale}>{locale}</option>
          ))}
        </Select>
      </Container>
    );
  }
}
