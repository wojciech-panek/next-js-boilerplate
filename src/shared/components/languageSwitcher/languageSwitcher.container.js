import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { withRouter } from 'next/router';

import withIntl from '../../../shared/lib/withIntl';
import { LanguageSwitcher } from './languageSwitcher.component';
import { selectLocalesLanguage } from '../../../modules/locales/locales.selectors';


const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({}, dispatch);

export default compose(
  withIntl,
  withRouter,
  connect(mapStateToProps, mapDispatchToProps),
)(LanguageSwitcher);
