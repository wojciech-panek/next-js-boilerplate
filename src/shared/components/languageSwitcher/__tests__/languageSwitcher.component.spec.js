import React from 'react';
import { expect } from 'chai';
import { shallow } from 'enzyme';
import { createSandbox } from 'sinon';

import { LanguageSwitcher } from '../languageSwitcher.component';
import { Select } from '../languageSwitcher.styles';
import { DEFAULT_LOCALE } from '../../../../i18n';
import routes from '../../../../routes';


describe('LanguageSwitcher: Component', () => {
  const sandbox = createSandbox();
  const defaultProps = {
    language: DEFAULT_LOCALE,
    router: { query: { lang: DEFAULT_LOCALE }, asPath: `/${DEFAULT_LOCALE}/some/custom/url` },
  };

  afterEach(() => {
    sandbox.restore();
  });

  const component = (props) => (
    <LanguageSwitcher {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should redirect after option click', () => {
    sandbox.stub(routes.Router, 'pushRoute');
    const wrapper = shallow(component());

    const event = { target: { value: 'not-default' } };
    wrapper.find(Select).prop('onChange')(event);

    expect(routes.Router.pushRoute).to.have.been.calledOnce;
    expect(routes.Router.pushRoute).to.have.been.calledWith('/not-default/some/custom/url');
  });
});
