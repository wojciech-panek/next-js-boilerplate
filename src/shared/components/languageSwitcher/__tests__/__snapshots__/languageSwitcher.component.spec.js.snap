// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher: Component should render correctly 1`] = `
<languageSwitcherstyles__Container>
  <languageSwitcherstyles__Select
    onChange={[Function]}
    value="en"
  >
    <option
      key="0"
      value="en"
    >
      en
    </option>
    <option
      key="1"
      value="pl"
    >
      pl
    </option>
  </languageSwitcherstyles__Select>
</languageSwitcherstyles__Container>
`;
